### PR TITLE
Make the Between filter easier to use and understand

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/25026
Demo 1 https://www.loom.com/share/c6e28955f0e848c097242009af55ef18
Demo 2 https://www.loom.com/share/8fcf1d1da8144545b4bc6c3ded7c3695

How to test:
- Open Orders table
- Click on `Created At` -> `Filter by this column` -> `Specific Dates`
- It should open `Between` tab with the date range from 30 days ago through today
- Make sure it is possible to change the date range by clicking on dates in the calendar and by typing in inputs

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/filters/pickers/DatePicker/RangeDatePicker.tsx [hunk 1]
```diff
@@ -1,82 +1,106 @@
...
+}
 
 export const BetweenPicker = ({
   className,
   filter: [op, field, startValue, endValue],
-  onFilterChange,
-  hideTimeSelectors,
   primaryColor,
+  hideTimeSelectors,
+  onFilterChange,
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/filters/pickers/DatePicker/RangeDatePicker.tsx:23]: Looks to me that this is just a props reordering. Not sure if this does anything.

**Review by WiNloSt — APPROVED** [review]
I know this is the current behavior, but I think after I select a start date on a between filter, the UI then focuses on the end date and shows today as a value. It kind of suggests that today is preselected, but that's not the case, and I can't even click "Add filter" as it's grayed out.
image


What do you think about making it look empty or even making it possible to add a filter without clicking on the end date?
image

FYI, @mazameli. Other than that, this PR addresses everything that's in the product doc.

**Comment by mazameli** [comment]
The `specific dates` state of the popover is annoyingly getting cropped off the side of my screen (window width = 1420px):

cropped-off-specific-dates

**Comment by mazameli** [comment]
Polish: it'd be a nice touch if, right after picking `specific dates`, when I'm defaulted into the Between tab if the first input field was in focus (i.e., my cursor was in it) so that I could start typing a date if I wanted to: 

image

**Comment by mazameli** [comment]
Beyond those two items, this works wonderfully! 👏🏼

**Pull request merged**